304-known-implementations.md

```diff
@@ -0,0 +1,123 @@
+
+The current document is written based on [release 0.0.2 of Signy][signy-release], which:
+
+- uses the [Notary][notary] client libraries, and communicates with a Notary server.
+- uses a slightly modified version of the [in-toto client libraries for Golang][in-toto-golang] to perform layout verification.
+- performs all in-toto verifications in a container based on a _verification image_ - this ensures the desired dependencies and their versions are available for all clients running the in-toto verifications, regardless of the operating system they are running. See [this project][in-toto-container] for an example of how verification images are used.
+- for thin CNAB bundles, signing will push the bundle to a compliant OCI registry, and verifying will pull the bundle from an OCI registry (the push and pull operations are performed using [`docker/cnab-to-oci`][cnab-to-oci]).
+- for thick CNAB bundles, Signy will pull the trust metadata from Notary, and will compare it against a local file (the thick bundle).s
+
+> [Notary][notary] is an opinionated implementation of the TUF specification written in Go, which has wide adoption in container registries, through the Docker Content Trust model.
+
+> Because Signy uses Notary, its implementation is very closely related to the Docker Content Trust model. For an architecture of a service running Notary, [see this document][notary-architecture].
+
+#### Using Signy to perform TUF verifications
+
+- signing and pushing a bundle:
+
+```
+$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 sign testdata/cnab/bundle.json localhost:5000/thin-bundle:v1
+INFO[0000] Pushed trust data for localhost:5000/thin-bundle:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] Starting to copy image cnab/helloworld:0.1.1
+INFO[0002] Completed image cnab/helloworld:0.1.1 copy
+INFO[0002] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/thin-bundle@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
+INFO[0002] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268"
+```
+
+- pulling from an OCI registry and verifying the signature of a bundle:
+
+```
+$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 verify localhost:5000/thin-bundle:v1
+INFO[0000] Pulled trust data for localhost:5000/thin-bundle:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] Pulling bundle from registry: localhost:5000/thin-bundle:v1
+INFO[0000] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+```
+
+#### Using Signy to also perform in-toto verifications
```

Comment:
Could we make it so that Signy automagically detects whether or not it needs to do in-toto verification? We could make this entire thing quite transparent to users.

Reply:
Oh, that's a really good idea.
Opened https://github.com/engineerd/signy/issues/38 to track it, thanks!